### ISSUE TITLE
NC20 - Unified search full width, not just pull-down

### ISSUE THREAD

**Issue by BloodyIron** [nextcloud/server#23328]
I upgraded to NC20 on Saturday, and while I am glad to see how many different things Unified search crosses, it now gives me far less results in the areas I care about, and far less efficiently.

I can understand the search pull-down for just seeing a few brief results works. But I want the "full screen" experience, similar to the previous version. But with the unified and "show more" experience.

When I want to see 50-100 results, doing that as it is in NC20 is painful and actually in a sense worse than pre-NC20. So can we please get this addressed in the near future? I don't really want to wait for NC21 for this to get addressed.

**Comment by thomasgg23** [comment]
Same thoughts as @BloodyIron took a screenshot from my server.

In my opinion it would be best to have the following workflow:
1) Enter a query and get the results in the "quick pull-down" search result area.
    Change the width of the search result area, to see more of the filenames.
2) Instead of "show more" and load a longer list of the search results in a small window/field, do a separate search result page where you show the results in "full size".
   So you can see the full filename and can scroll in the same window. Would be much easier to explain to the user.

So a fast search with less results in the common search field and a for a deeper search a new window with full listing.

![search_field](https://user-images.githubusercontent.com/2633051/96333339-8db51f00-1069-11eb-8b00-1a8da2db0deb.jpg)

**Comment by skjnldsv** [comment]
Yes, this was actually a discussion about a future enhancement.
A bigger Modal with more results details/preview :)
cc @jancborchardt

**Comment by BloodyIron** [comment]
I'd prefer it transition to not a modal and just be the full width as previous if you want more results, and then show a lot more results. The pull-down menu is handy if you can quickly find what you want inside a limited set of results, but when you have very many results, a modal can also be clunky and break the UX.

**Comment by skjnldsv** [comment]
That is not the planned. We'll definitely keep the current drop-down behaviour.

But if that request came to be implemented, it would be with an option to enlarge the results and gives enhanced details on the results (files details, contact info, calendar event details...) Similar to how MacOS is currently doing with its spotlight search :)

**Comment by BloodyIron** [comment]
Quite frankly if I could turn the drop-down off completely I would. Right now, I'm on 2560x1440 resolution for my monitor, and that pull-down menu takes up about 5% of the usable space on my screen. This is extremely inefficient, and a very substantial reduction in my ability to actually search and find the things I'm looking for (vs NC18/19).

Of course "this is not planned", that's why we're talking about it.

Look, I don't have the exact numbers right now, but I have a very large amount of files to search through, and generally that's how nextCloud is used. I don't want to have to deal with a small fraction of my usable space to find far less results than I was getting before.

Previously, with NC18/19 I was able to get about 50 results per search in the middle pane. Granted, I had no way of expanding that to get more results (which I did want), however the way that is done now is actually worse than before.

Now I get about 3-4 results per result type, and expanding it each time maybe gives me 5-10 more results.

I want results in the range of about 200 or so.

THIS DOES NOT WORK FOR ME and I'm confident it is worse for many others too. As it is now, I barely want to use the search because it's so useless to me, whereas previously it was a critical part to my regular usage of nextCloud.

It's not all salt and vinegar though, I do anticipate benefiting from the search expansion into other areas (like GitLab, etc), but that's a future benefit, and right now I'm talking about the significant usability reduction that is the pull-down menu and how search works today. It sucks.

**Comment by skjnldsv** [comment]
> THIS DOES NOT WORK FOR ME and I'm confident it is worse for many others too. As it is now, I barely want to use the search because it's so useless to me, whereas previously it was a critical part to my regular usage of nextCloud.
> 
> It's not all salt and vinegar though, I do anticipate benefiting from the search expansion into other areas (like GitLab, etc), but that's a future benefit, and right now I'm talking about the significant usability reduction that is the pull-down menu and how search works today. It sucks.

Sorry this doesn't fit your needs. We can't please everyone of course.
We've also receive tons of feedback for people that were perfectly happy with it.
On another note, please reconsider using sentences like "it sucks." as I'm positive it will just make developer ignore your issue. I'll recommand [you read our code of conduct](https://nextcloud.com/contribute/code-of-conduct/) and think about how your wording make it seems you're feeling entitled to changes on a free opensource software.
That is definitely not how I'm used to collaborate.


___
> THIS DOES NOT WORK FOR ME

> It sucks.

> But I want the

> So can we please get this addressed in the near future? I don't really want to wait for NC21 for this to get addressed.

**Comment by jospoortvliet** [comment]
@skjnldsv I agree, having a bigger/full screen view would be helpful. But isn't there a role for apps, too? I like how the app store filters in the main view based on the search terms - and I'd love to see the same in files and other apps. Though maybe that interaction can be made smarter - it's confusing to have the popup and the main window view if what you want is one of those. But that's probably not super easy, I don't have a good idea for how to do that. a fun challenge for our designers ;-)

Maybe it'd already help if below each result the 'more results' button opens the specified app with a full screen view? But that would need app work. Pffft.
